multistream-select/src/error.rs

```diff
@@ -49,3 +51,36 @@ impl From<IoError> for ProtocolChoiceError {
 		MultistreamSelectError::from(err).into()
 	}
 }
+
+impl error::Error for ProtocolChoiceError {
+	#[inline]
+	fn description(&self) -> &str {
+		match *self {
+			ProtocolChoiceError::MultistreamSelectError(_) => {
+				"error in the protocol"
+			},
+			ProtocolChoiceError::UnexpectedMessage => {
+				"received a message from the remote that makes no sense in the current context"
+			},
+			ProtocolChoiceError::NoProtocolFound => {
+				"we don't support any protocol in common with the remote"
+			},
+		}
+	}
+
+	fn cause(&self) -> Option<&error::Error> {
+		match *self {
```

Comment:
Style nit: maybe use an `if let`?

Reply:
Nit: A match is better if we add more variants in the future